nifi/nifi-nar-bundles/nifi-flume-bundle/nifi-flume-processors/src/test/java/org/apache/nifi/processors/flume/FlumeSourceProcessorTest.java

```diff
@@ -0,0 +1,140 @@
+    @Test
+    public void testValidators() {
+        TestRunner runner = TestRunners.newTestRunner(FlumeSourceProcessor.class);
+        Collection<ValidationResult> results;
+        ProcessContext pc;
+
+        results = new HashSet<>();
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(1, results.size());
+        for (ValidationResult vr : results) {
+            logger.error(vr.toString());
+            Assert.assertTrue(vr.toString().contains("is invalid because Source Type is required"));
+        }
+
+        // non-existent class
+        results = new HashSet<>();
+        runner.setProperty(FlumeSourceProcessor.SOURCE_TYPE, "invalid.class.name");
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(1, results.size());
+        for (ValidationResult vr : results) {
+            logger.error(vr.toString());
+            Assert.assertTrue(vr.toString().contains("is invalid because unable to load source"));
+        }
+
+        // class doesn't implement Source
+        results = new HashSet<>();
+        runner.setProperty(FlumeSourceProcessor.SOURCE_TYPE, NullSink.class.getName());
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(1, results.size());
+        for (ValidationResult vr : results) {
+            logger.error(vr.toString());
+            Assert.assertTrue(vr.toString().contains("is invalid because unable to create source"));
+        }
+
+        results = new HashSet<>();
+        runner.setProperty(FlumeSourceProcessor.SOURCE_TYPE, AvroSource.class.getName());
+        runner.enqueue(new byte[0]);
+        pc = runner.getProcessContext();
+        if (pc instanceof MockProcessContext) {
+            results = ((MockProcessContext) pc).validate();
+        }
+        Assert.assertEquals(0, results.size());
+    }
+
+    @Test
+    public void testSequenceSource() {
+        TestRunner runner = TestRunners.newTestRunner(FlumeSourceProcessor.class);
+        runner.setProperty(FlumeSourceProcessor.SOURCE_TYPE, "seq");
+        runner.run();
+        List<MockFlowFile> flowFiles = runner.getFlowFilesForRelationship(FlumeSourceProcessor.SUCCESS);
+        Assert.assertEquals(1, flowFiles.size());
+        for (MockFlowFile flowFile : flowFiles) {
+          logger.error(flowFile.toString());
+            Assert.assertEquals(1, flowFile.getSize());
+        }
+    }
+
+    @Test
+    public void testSourceWithConfig() throws IOException {
+        File spoolDirectory = new File("target/spooldir");
```

Comment:
please get this from junit / maven. For example, by using a TemporaryDirectory JUnit rule.

    @Rule
    public final TemporaryFolder temp = new TemporaryFolder();

    @Test
    public void someTest() {
        File destDir = temp.newFolder("spooldir");